packages/platform-sdk-generator/src/model/Type.ts

```diff
@@ -22,7 +22,9 @@ export abstract class Type {
   abstract getTsReferenceString(localNamespace: string | undefined): string;
 
   getPropertyDeclaration(name: string, localNamespace: string | undefined) {
-    return `${name}: ${this.getDeclaration(localNamespace)}`;
+    return `${name}: ${
```

Comment:
This part of this change makes me a little worried. I am guessing that changing `Component`'s `getTsReferenceString` to check for `Record` might be the more stable change here. I think you can then also replace the change to `Component`'s `getDeclaration` to just call its own `getTsReferenceString` when getting the name.

Reply:
Done